### PR TITLE
Add non-record 4xH100 10L Int5-MLP submission

### PR DESCRIPTION
**Pull request by ReNothingg**
# Non-record: 10L Int5-MLP on 4xH100 for 1200s

This is a non-record submission based on the 10L Int5-MLP recipe, rerun on 4xH100.

It is not intended for the main 10-minute 8xH100 leaderboard. This run uses 4 GPUs with the script's built-in gradient accumulation behavior and a 1200-second wallclock cap.

## Result

- val_loss: 1.92858953
- val_bpb: 1.14222237
- bytes_total: 15803072
- stopping_early: train_time=1200141ms step=6847/20000
- final_eval_mode: sliding_window stride=64

## Command

```bash
SEED=1337 \
RUN_ID=int5_4gpu_seed1337_1200s_20260324_055951 \
DATA_PATH=/workspace/parameter-golf/data/datasets/fineweb10B_sp1024 \
TOKENIZER_PATH=/workspace/parameter-golf/data/tokenizers/fineweb_1024_bpe.model \
MAX_WALLCLOCK_SECONDS=1200 \
torchrun --standalone --nproc_per_node=4 train_gpt.py
```

## Notes

- Final artifact used zstd compression.
- This submission is for the non-record track.

### CODE REVIEW

**Comment by MatoTeziTanka** [comment]
### Community Review — Add non-record 4xH100 10L Int5-MLP submission

**BPB:** 1.1422 | **Compliance:** LOOKS CLEAN — pure-neural submission, no TTT/SLOT/n-gram-cache

**What I found in the code** (head SHA `3455bb2f10d2`, file `records/track_non_record_16mb/2026-03-24_4xH100_20min_10L_Int5MLP_seed1337/train_gpt.py`):

Static code review found no TTT adaptation function, no SLOT optimization loop, no n-gram-cache class, and no pre-quant val-token fine-tune. The eval path uses the standard sliding-window stride-64 pattern. The submission is a pure-neural architecture iteration on the standard SP1024/SP4096/SP8192 baseline.

**CPU smoke test (CT2038 proteus-engine, 2026-04-11):** import OK in 0.04s, dim=512, layers=10, vocab=1024, code=52930 B, SMOKE_TEST_PASS

**Verdict:** LOOKS CLEAN.

**Recommendation to @cocohearts @valerio-oai @0hq @yuzhougu-oai @notapplica:** **MERGE** pending the usual record-track checks (3-seed validation, under-16MB artifact cap, ≤600s train + ≤600s eval on 8×H100 SXM). No compliance flags from the classification pass — this looks like a clean pure-neural iteration on the standard baseline.

*Auto-classification caveat:* this review was drafted by the AST-based classifier. If there's a non-standard eval mechanism (logit postprocessing, hedge mixing, etc.) that I missed because it's factored into a helper file or a non-standard function name, please flag it and I'll re-run the audit manually.

---
*Reviewed by [@MatoTeziTanka](https://github.com/MatoTeziTanka) — [The Agora](https://matotezitanka.github.io/parameter-golf). CPU smoke test (CT2038 proteus-engine, 2026-04-11): import OK in 0.04s, dim=512, layers=10, vocab=1024, code=52930 B, SMOKE_TEST_PASS. Classification via deterministic AST-based `classify_prs.py` (pattern bank derived from ~65 manually-reviewed PRs earlier in the 2026-04-11 sweep). This review was auto-drafted from a template and spot-checked before posting — if the template misread your code, please call it out so I can iterate the classifier.*
